tensorflow/tools/ci_build/builds/libtensorflow.sh

```diff
@@ -78,9 +78,40 @@ function build_libtensorflow_tarball() {
     //tensorflow/tools/lib_package:libtensorflow_proto.zip
 
   mkdir -p ${DIR}
+
   cp bazel-bin/tensorflow/tools/lib_package/libtensorflow.tar.gz ${DIR}/libtensorflow${TARBALL_SUFFIX}.tar.gz
   cp bazel-bin/tensorflow/tools/lib_package/libtensorflow_jni.tar.gz ${DIR}/libtensorflow_jni${TARBALL_SUFFIX}.tar.gz
-  cp bazel-bin/tensorflow/java/libtensorflow.jar bazel-bin/tensorflow/java/libtensorflow-src.jar ${DIR}
+  cp bazel-bin/tensorflow/java/libtensorflow.jar ${DIR}
+  cp_normalized_srcjar bazel-bin/tensorflow/java/libtensorflow-src.jar ${DIR}/libtensorflow-src.jar
   cp bazel-genfiles/tensorflow/tools/lib_package/libtensorflow_proto.zip ${DIR}
   chmod -x ${DIR}/*
 }
+
+# Helper function to copy a srcjar after moving any source files
+# directly under the root to the "maven-style" src/main/java layout
+# Arguments:
```

Comment:
I'm a bit lost on why we end up in this situation (that the src_jar contains anything outside the "maven-style" hierarchy. Perhaps that will dawn on my as I see the rest of the CL, just wondering out aloud for now.

Reply:
The problem is that the annotation processor doesn't control where its generated sources are written. The processor only provides a package name - the rest of it is determined by the environment.

The environment is first determined by the way bazel calls javac, and then by how bazel packages such generated files into its -srcs.jar. bazel generates files into a bazel-out directory, and then jars them directly in the root of its -srcs.jar file. Is there an option I missed to control all this? I tried adding things like compiler options, but this ends up just confusing bazel.

Reply:
Thanks for the review Asim!

Have added a comment here.